jnigen/lib/src/config/yaml_reader.dart

```diff
@@ -69,97 +67,58 @@ class YamlReader {
         throw ConfigException('override does not match expected pattern');
       }
     }
+    final config = cli_config.Config.fromConfigFileContents(
+      commandLineDefines: results['override'],
+      workingDirectory: Directory.current.uri,
+      environment: Platform.environment,
+      fileContents: configFileContents,
+      fileSourceUri: configFileUri,
+    );
     return YamlReader.of(
-        properties, yamlMap, configFile != null ? File(configFile) : null);
+      config,
+      configFileUri?.resolve('.'),
+    );
   }
 
-  bool? getBool(String property) {
-    if (cli.containsKey(property)) {
-      final v = cli[property]!;
-      if (v == 'true') {
-        return true;
-      }
-      if (v == 'false') {
-        return false;
-      }
-      throw ConfigException('expected boolean value for $property, got $v');
-    }
-    return getYamlValue<bool>(property);
-  }
+  bool? getBool(String property) => _config.optionalBool(property);
 
-  String? getString(String property) {
-    final configValue = cli[property] ?? getYamlValue<String>(property);
-    return configValue;
-  }
+  String? getString(String property) => _config.optionalString(property);
 
   /// Same as [getString] but path is resolved relative to YAML config if it's
   /// from YAML config.
-  String? getPath(String property) {
-    final cliOverride = cli[property];
-    if (cliOverride != null) return cliOverride;
-    final path = getYamlValue<String>(property);
-    if (path == null) return null;
-    // In (very unlikely) case YAML config didn't come from a file,
-    // do not try to resolve anything.
-    if (yamlFile == null) return path;
-    final yamlDir = yamlFile!.parent;
-    return yamlDir.uri.resolve(path).toFilePath();
-  }
+  String? getPath(String property) =>
+      _config.optionalPath(property)?.toFilePath();
 
-  List<String>? getStringList(String property) {
-    final configValue = cli[property]?.split(';') ??
-        getYamlValue<YamlList>(property)?.cast<String>();
-    return configValue;
-  }
+  List<String>? getStringList(String property) => _config.optionalStringList(
+        property,
+        splitCliPattern: ';',
+        combineAllConfigs: false,
+      );
 
   List<String>? getPathList(String property) {
-    final cliOverride = cli[property]?.split(';');
-    if (cliOverride != null) return cliOverride;
-    final paths = getYamlValue<YamlList>(property)?.cast<String>();
-    if (paths == null) return null;
-    // In (very unlikely) case YAML config didn't come from a file.
-    if (yamlFile == null) return paths;
-    final yamlDir = yamlFile!.parent;
-    return paths.map((path) => yamlDir.uri.resolve(path).toFilePath()).toList();
+    final configResult = _config.optionalPathList(
+      property,
+      combineAllConfigs: false,
+      splitCliPattern: ';',
+    );
+    return configResult?.map((e) => e.path).toList();
   }
 
   String? getOneOf(String property, Set<String> values) {
```

Comment:
I think getOneOf can be moved to `package:config`, or a more structured one like `getEnum<T extends Enum>`.